code/modules/mob/living/silicon/ai/life.dm

```diff
@@ -64,7 +64,6 @@
 	if(!fire_res_on_core)
 		health -= getFireLoss()
 	update_stat()
-	diag_hud_set_status()

```

Comment:
Moved from updatehealth() to updatestat() for the same reason as explained a bit above.